android/java/org/chromium/chrome/browser/crypto_wallet/fragments/dapps/SignTransactionFragment.java

```diff
@@ -0,0 +1,364 @@
+                    return;
+                }
+                break;
+            default: // Do nothing
+        }
+        mTxRequestNumber++;
+    }
+
+    private void updateActionState(boolean isEnabled) {
+        if (mSignTxStep == SignTx.SIGN_RISK) return;
+        mBtCancel.setEnabled(isEnabled);
+        mBtSign.setEnabled(isEnabled);
+        if (isEnabled) {
+            mBtSign.setBackgroundTintList(ColorStateList.valueOf(
+                    ContextCompat.getColor(requireContext(), R.color.brave_action_color)));
+
+        } else {
+            mBtSign.setBackgroundTintList(ColorStateList.valueOf(
+                    ContextCompat.getColor(requireContext(), R.color.modern_grey_700)));
+        }
+    }
+
+    private void fetchSignRequestData() {
+        switch (mActivityType) {
+            case SIGN_TRANSACTION:
+                mWalletModel.getDappsModel().fetchSignTxRequest().observe(
+                        getViewLifecycleOwner(), requests -> {
+                            if (requests.size() == 0) return;
+                            mSignTransactionRequests = requests;
+                            updateSignDataAndDetails();
+                        });
+                break;
+            case SIGN_ALL_TRANSACTIONS:
+                mWalletModel.getDappsModel().fetchSignAllTxRequest().observe(
+                        getViewLifecycleOwner(), requests -> {
+                            if (requests.size() == 0) return;
+                            mSignAllTransactionRequests = requests;
+                            updateSignDataAndDetails();
+                        });
+                break;
+            default: // Do nothing
+        }
+    }
+
+    private void updateTxPanelPerStep() {
+        if (mSignTxStep == SignTx.SIGN_RISK) {
+            mViewPager.setVisibility(View.GONE);
+            mTvTxCounter.setVisibility(View.GONE);
+            mBtnCounterNext.setVisibility(View.GONE);
+            mWarningLl.setVisibility(View.VISIBLE);
+            mBtSign.setBackgroundTintList(ColorStateList.valueOf(
+                    ContextCompat.getColor(requireContext(), R.color.brave_theme_error)));
+            mBtSign.setCompoundDrawablesWithIntrinsicBounds(0, 0, 0, 0);
+            mBtSign.setText(R.string.continue_text);
+        } else if (mSignTxStep == SignTx.SIGN_TX) {
+            mWarningLl.setVisibility(View.GONE);
+            mViewPager.setVisibility(View.VISIBLE);
+            mBtSign.setBackgroundTintList(ColorStateList.valueOf(
+                    ContextCompat.getColor(requireContext(), R.color.brave_action_color)));
+            mBtSign.setCompoundDrawablesRelativeWithIntrinsicBounds(R.drawable.ic_key, 0, 0, 0);
+            mBtSign.setText(R.string.brave_wallet_sign_message_positive_button_action);
+        }
+    }
+
+    private void processRequest(boolean isApproved) {
+        if (isApproved) {
+            // Continue clicked, update ui
+            if (mSignTxStep == SignTx.SIGN_RISK) {
```

Comment:
could be moved under on `if`

Reply:
updated: https://github.com/brave/brave-core/pull/15389/commits/0c925e12fee38080c6ba602895d813a2d98e489d